addons/account/models/product.py

```diff
@@ -156,3 +162,8 @@ def _get_tax_included_unit_price(self, company, currency, document_date, documen
             product_price_unit = product_currency._convert(product_price_unit, currency, company, document_date)
 
         return product_price_unit
+
+    def _compute_tax_string(self):
+        for record in self:
+            record.tax_string = record.product_tmpl_id._construct_tax_string(record, record.lst_price)
```

Comment:
This tracebacks, and it is missing dependencies

Reply:
> Please mark the PR as ready so that I can approve it when you are finished with the last changes :)

well, my fear is that because now `_construct_tax_string` is an instance method and the inheritance is delegation (so polymorphism won't work), it will be very confusing later when extending this. for now it is not an issue because the only attribute used to calculate the tax string is the price but imagine wanting to add other variables that might differ between product.product and product.template.

 tl;dr
`self` in `_construct_tax_string` is always `product_tmpl_id` regardless of where it was called from

Reply:
also why the style check fails with C0305, I added one line at the end

https://github.com/odoo/odoo/pull/87248/files#diff-6193fe8fb1a4b291ce7c82ef509968e9b381312909738d621adda02b2177807fR169

Reply:
> also why the style check fails with C0305, I added one line at the end

You have 2 end of line char instead of one, leading in one blank/useless line

> `self` in `_construct_tax_string` is always `product_tmpl_id` regardless of where it was called from

How is that answering any of my remarks?
And why is that an issue?

Reply:
I did fix the errors, I don't know what dependencies it misses.

this was another point.

the issue is that the intuitive behavior when someone sees this implementation will be that this is a normal case of polymorphism and can thus be extended further which is not the case